### PR TITLE
Wouter entity in engine

### PR DESCRIPTION
**Pull request by WoutervanGeel**

### CODE REVIEW

**Comment by WoutervanGeel** [comment]
Good point. in the old architecture this was the idea. This was to let tileobjects control all action within itself. This would be separation of concern.

That being said, that is the old architecture. In this new architecture this is completely unnecessary. I will rewrite the code. thanks for the feedback

**Comment by ThomasvLingen** [comment]
Cool, cool. Hit me up when you want me to review the code!

File: src/engine/ImageTypes.hpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#define MINOR_VIDYA_IMAGETYPES_H
+
+#include <vector>
+#include <map>
+
+namespace Engine {
+    using std::vector;
+    using std::map;
+
+    typedef std::vector<uint32_t> ImageBuffer;
```
Comment (ThomasvLingen) [inline src/engine/ImageTypes.hpp]: this can be just `vector`

File: src/engine/domain/Entity.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    /// \brief Destructor of the class
+    Entity::~Entity()
+    {
+    }
+
+    /// \brief Getter for the position of the entity
+    /// \return returns a CoordinateInt position
+    CoordinateInt& Entity::get_position()
+    {
+        return _position;
```
Comment (ThomasvLingen) [inline src/engine/domain/Entity.cpp]: use `this->`

File: src/engine/domain/Entity.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    /// \brief Getter for the image buffer of the entity
+    /// \return returns an ImageBuffer
+    ImageBuffer& Entity::get_texture()
+    {
+        return _texture;
```
Comment (ThomasvLingen) [inline src/engine/domain/Entity.cpp]: same here

File: src/engine/domain/Entity.hpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+#include "../raycasting/RayCastingTypes.hpp"
+#include "../ImageTypes.hpp"
+
+namespace Engine {
+    class Entity {
+
+    public:
+        Entity(ImageBuffer& texture, CoordinateInt position);
+        ~Entity();
```
Comment (ThomasvLingen) [inline src/engine/domain/Entity.hpp]: Prefer to make constructors virtual by default.

Saves us a lot of potential trouble when deriving something from this class

File: src/engine/ImageTypes.hpp (hunk 1 above)
Comment (WoutervanGeel) [inline src/engine/ImageTypes.hpp]: Done

File: src/engine/domain/Entity.cpp (hunk 2 above)
Comment (WoutervanGeel) [inline src/engine/domain/Entity.cpp]: Done

File: src/engine/domain/Entity.cpp (hunk 3 above)
Comment (WoutervanGeel) [inline src/engine/domain/Entity.cpp]: Done

File: src/engine/domain/Entity.hpp (hunk 4 above)
Comment (WoutervanGeel) [inline src/engine/domain/Entity.hpp]: Bedoel je niet alleen destructor?

File: src/engine/domain/World.hpp [hunk 5]
```diff
@@ -23,6 +27,8 @@ namespace Engine {
         virtual TileObject* get_tile(CoordinateInt coordinate) = 0;
         virtual void handle_input(Input keys) = 0;
 
+        vector<Entity> get_entities();
```
Comment (ThomasvLingen) [inline src/engine/domain/World.hpp]: Oh, another minor thing. Do we want to return the vector here?
Why not a vector&? Saves us copying x Entities.

**Pull request merged**